src/ScaffoldInstallerPlugin.php

```diff
@@ -468,10 +468,26 @@ function($key) {
         // Add Redis service.
         if (file_exists('./.ddev/docker-compose.redis.yaml')) {
             $redisConfig = Yaml::parseFile('.ddev/docker-compose.redis.yaml');
-            $redisImage = explode(':',
-                $redisConfig['services']['redis']['image']);
+            $imageRaw = $redisConfig['services']['redis']['image'] ?? '';
+
+            // Normalize image from possible environment syntax.
+            if (preg_match('/^\$\{[^:}]+:-(.+)\}$/', $imageRaw, $matches)) {
+                $image = $matches[1];
+            } elseif (preg_match('/^([^:]+):\$\{[^:}]+:-(.+)\}$/', $imageRaw, $matches)) {
+                $image = "{$matches[1]}:{$matches[2]}";
+            } else {
+                $image = $imageRaw;
+            }
```

Comment:
This could be simplified I think. If you change `$imageRaw = ` above to `$image`, then you can delete this else statement too?